### PR TITLE
Add security considerations page

### PR DESCRIPTION
**Pull request by pvanderknyff**
First stab at adding the Security Considerations page. We want to document the common security reasons we reject connectors from the Exchange.

### CODE REVIEW

**Review by smarvin — COMMENTED** [review]
There's an extra period in the last sentence of line 14 ("The values of every non-secure UI field on the connection dialog will be logged in plain text by Tableau as well as persisted in easily-inspectable XML in Tableau workbooks, datasources and Prep flows..")

**Review by joe-langford — APPROVED** [review]

File: docs/security-considerations.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+We will also pull any connector from the Tableau Exchange that we find a security vulnerability in, and that vulnerability must be addressed before the connector can go back to the Exchange.
+
+
+# Secure Attributes on the Connection Dialog
+
+The Connector SDK allows the connection dialog to be customized, using platform-defined fields as well as the creation of new, vendor-specific fields. It is important to note that these fields, with the exception of the built-in password field, are not secure. **The values of every non-secure UI field on the connection dialog will be logged in plain text by Tableau as well as persisted in easily-inspectable XML in Tableau workbooks, datasources and Prep flows.**
+
+At the moment, the only field that can be marked secure is password. This means that the values customers enter for vendor-specific fields **must not contain secrets, credentials or personally identifiable information (PII),** and connectors with such fields will not be approved for the Tableau Exchange. This is a known limitation of the SDK, and we are looking at ways to mitigate it in the long term.
```
Comment (rosswbrown) [inline docs/security-considerations.md:16]:
```suggestion
Currently the only field that can be marked secure is `password`. This means that the values customers enter for vendor-specific fields **must not contain secrets, credentials or personally identifiable information (PII),** and connectors with such fields will not be approved for the Tableau Exchange. This is a known limitation of the SDK and we are looking at ways to remove this constraint in the long term.
```

File: docs/security-considerations.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Examples of problematic UI fields that will be rejected:
+- Proxy Password
+- OAuth secrets (Use the official [Tableau OAuth flow](({{ site.baseurl }}/docs/oauth)) instead)
+- **Arbitrary JDBC Properties** (See following section)
+
+## Arbitrary Driver Properties
```
Comment (rosswbrown) [inline docs/security-considerations.md:23]: Is freeform a better term? I can't decide.

File: docs/security-considerations.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+We recommend the following workarounds instead of an "Arbitrary Driver Properties" field:
+- To unblock users needing to set non-sensitive driver properties, add them as separate fields on the Advanced tab using [Connection Dialogs V2 framework](({{ site.baseurl }}/docs/mcd))
+- For JDBC, Customers can set driver properties using [JDBC Properties files](https://kb.tableau.com/articles/howto/Customizing-JDBC-Connections)
+
+# Driver Vulnerabilities
+
+The Tableau Exchange does not host drivers, and the responsibility of ensuring the driver is secure falls onto the driver authors.
+
+However, for the connectors that are hosted in Tableau Online, Tableau will need to host the driver in our environments. In these cases, the driver will be scanned for security vulnerabilities, including out-of-date and vulnerable third-party libraries.
```
[the remark this answers is not in the document]
Comment (pvanderknyff) [reply docs/security-considerations.md]: Already updated based on suggestion in slack thread.

File: docs/security-considerations.md (hunk 2 above)
Comment (pvanderknyff) [reply docs/security-considerations.md:23]: I do like freeform better. It's clearer.

**Review by rosswbrown — APPROVED** [review]

**Review by VasanthiNeelagiri — APPROVED** [review]
Looks good to me. Thanks for putting in the effort Patrick.

**Pull request merged**